### PR TITLE
Fix matrix plotting labels

### PR DESCRIPTION
**Pull request by mjboos**
Fixes issues #2049 i.e. `plot_matrix` breaking if an empty list is supplied as the labels argument and `auto_fit=True`.

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
Looks good, but can we have a test: something that used to break and now works.

**Comment by mjboos** [comment]
Fair point, I've added two tests.

File: nilearn/plotting/matrix_plotting.py [hunk 1]
```diff
@@ -80,8 +80,12 @@ def plot_matrix(mat, title=None, labels=None, figure=None, axes=None,
         display : instance of matplotlib
             Axes image.
     """
+    # we need a list so an empty one will be cast to False
+    if isinstance(labels, np.ndarray):
+        labels = labels.tolist()
+
     if reorder:
-        if labels is None or labels is False:
+        if not labels:
```
Comment (jeromedockes) [inline nilearn/plotting/matrix_plotting.py]: I think iit is better to explicitly check if labels is `None` and if not use `len` or a numpy array's `size` if you need to check if it is empty rather than the implict cast to `bool`.

it would also make sense to check that if labels is not None it must be a sequence of the same size as the plotted matrix number of rows

Comment (mjboos) [reply nilearn/plotting/matrix_plotting.py]: The first point might be a matter of style, but [PEP8](https://www.python.org/dev/peps/pep-0008/) recommends exactly that - casting the list to `bool`.

I implemented a check for the length of labels, good idea.

Comment (kchawla-pi) [reply nilearn/plotting/matrix_plotting.py]: If `labels` is a pure Python list/tuple then `not labels` works fine. However if it's a numpy array, then the cast to bool is ambiguous & will raise a warning.
If it's possible this will be a numpy array then `is None` & `is False` should be used.

Comment (jeromedockes) [reply nilearn/plotting/matrix_plotting.py]: the doc doesn't list `False` or an empty list as valid inputs. let's make the docstring more explicit that it should be `None` for no labels, or a sequence of length number of rows in the matrix, raise an error if the input is not one of these two things, build an `ndarray` from the labels if they are not None. and use explicit `is None` when needed.

Comment (mjboos) [reply nilearn/plotting/matrix_plotting.py]: @kchawla-pi it isn't possible that it's a numpy array, and a test is implemented that checks for the correct handling of an ndarray as input
@jeromedockes fair point, the documentation is ambiguous. The motivating example for this PR was an error that arose when `labels=[]`, I think it makes sense to cover that case, keep the implicit casting to bool, then make the docs more explicit. IMO these are all reasonable user inputs and casting to bool makes the code more readable, even if it is somewhat less explicit

Comment (jeromedockes) [reply nilearn/plotting/matrix_plotting.py]: this is true. it just feels a bit awkward having to cast the array to list just evaluate if its empty and then back to array, but it does allow handling None, False and an empty sequence uniformly. let's keep it the way you wrote it!

**Comment by jeromedockes** [comment]
@mjboos the circleci failure should be fixed by merging (or rebasing on) master

**Comment by mjboos** [comment]
> @mjboos the circleci failure should be fixed by merging (or rebasing on) master

I did rebase on master, but it still fails unfortunately

**Comment by jeromedockes** [comment]
> I did rebase on master, but it still fails unfortunately

yest but it's a different failure, this time its an internal error in the server from which we download the development dataset. i restarted it

File: nilearn/plotting/tests/test_matrix_plotting.py [hunk 2]
```diff
@@ -21,6 +21,14 @@ def test_matrix_plotting():
     # test if it returns an AxesImage
     ax.axes.set_title('Title')
     plt.close()
+    # test if an empty list works as an argument for labels
+    ax = plot_matrix(mat, labels=[])
```
Comment (jeromedockes) [inline nilearn/plotting/tests/test_matrix_plotting.py]: could we also test with `labels=None` (the default)?

**Comment by kchawla-pi** [comment]
Hi @mjboos we are so close to greatness! Would you be able to complete the changes & then we can merge!

**Comment by kchawla-pi** [comment]
Add it to the whats_new under Fixes.

File: nilearn/plotting/tests/test_matrix_plotting.py (hunk 2 above)
Comment (mjboos) [reply nilearn/plotting/tests/test_matrix_plotting.py]: Sure, added the test.

**Comment by kchawla-pi** [comment]
@mjboos The failure is related to conda, not changes made by you,  and I will file a fix for that and update you here.

File: nilearn/plotting/tests/test_matrix_plotting.py (hunk 2 above)
Comment (jeromedockes) [reply nilearn/plotting/tests/test_matrix_plotting.py]: thanks!

**Comment by jeromedockes** [comment]
+1 for merge

**Comment by kchawla-pi** [comment]
Whats_new fixes entry pending. THen we can merge.

**Comment by mjboos** [comment]
Okay, added it to what's new.
